server/routerlicious/packages/services-shared/src/runner.ts

```diff
@@ -31,11 +31,18 @@ export async function run<T extends IResources>(
 	const customizations = await (resourceFactory.customize
 		? resourceFactory.customize(config)
 		: undefined);
-	const resources = await resourceFactory.create(config, customizations);
-	const runner = await runnerFactory.create(resources);
+	const resources = await resourceFactory.create(config, customizations).catch((error) => {
+		error.errorLabel = `resourceFactory:create:${error.errorLabel ?? ""}`;
```

Comment:
For all of these `catch` statements, we're assuming `error` exists and is an `"object"`, when it could _technically_ be anything (`unknown` type). So it's unsafe to assign and read like this.

Maybe add a helper function like

```ts
/**
 * Prefixes a caught error if possible.
 * ```ts
 * asyncCall().catch((error: unknown) => {
 *   prefixErrorLabel(error, "my:prefix");
 *   log("AsyncCall Failed", error);
 * });
 * // Example Result: "AsyncCall Failed" { errorLabel: "my:prefix" }
 * ```
 */
function prefixErrorLabel(error: unknown, prefix: string): void {
  if (typeof error !== "object") return;
  const errorLabel = [prefix];
  if (error.errorLabel) {
    errorLabel.push(error.errorLabel);
  }
  error.errorLabel = errorLabel.join(":");
}
```

Reply:
Thanks for the great catch, Zach. I just updated the code.